components/movementsensor/mpu6050/mpu6050_test.go

```diff
@@ -178,3 +179,129 @@ func TestSuccessfulInitializationAndClose(t *testing.T) {
+		Name:  "movementsensor",
+		Model: model,
+		Type:  movementsensor.SubtypeName,
+		ConvertedAttributes: &AttrConfig{
+			BoardName:              testBoardName,
+			I2cBus:                 i2cName,
+			UseAlternateI2CAddress: true,
+		},
+	}
+
+	i2cHandle := &inject.I2CHandle{}
+	i2cHandle.ReadBlockDataFunc = func(ctx context.Context, register byte, numBytes uint8) ([]byte, error) {
+		if register == defaultAddressRegister {
+			return []byte{expectedDefaultAddress}, nil
+		}
+		return mockData, nil
+	}
+	i2cHandle.WriteByteDataFunc = func(ctx context.Context, b1, b2 byte) error {
+		return nil
+	}
+	i2cHandle.CloseFunc = func() error { return nil }
+	mockBoard := &inject.Board{}
+	mockBoard.I2CByNameFunc = func(name string) (board.I2C, bool) {
+		i2c := &inject.I2C{}
+		i2c.OpenHandleFunc = func(addr byte) (board.I2CHandle, error) {
+			return i2cHandle, nil
+		}
+		return i2c, true
+	}
+	return cfg, registry.Dependencies{
+		resource.NameFromSubtype(board.Subtype, testBoardName): mockBoard,
+	}
+}
+
+//nolint:dupl
+func TestLinearAcceleration(t *testing.T) {
+	linearAccelMockData := make([]byte, 16)
+	// x-accel
+	linearAccelMockData[0] = 64
+	linearAccelMockData[1] = 0
+	expectedAccelX := 9810.0
+	// y-accel
+	linearAccelMockData[2] = 32
+	linearAccelMockData[3] = 0
+	expectedAccelY := 4905.0
+	// z-accel
+	linearAccelMockData[4] = 16
+	linearAccelMockData[5] = 0
+	expectedAccelZ := 2452.5
+
+	logger := golog.NewTestLogger(t)
+	cfg, deps := setupDependencies(linearAccelMockData)
+	sensor, err := NewMpu6050(context.Background(), deps, cfg, logger)
+	test.That(t, err, test.ShouldBeNil)
+	defer utils.TryClose(context.Background(), sensor)
+	testutils.WaitForAssertion(t, func(tb testing.TB) {
+		linAcc, err := sensor.LinearAcceleration(context.Background(), nil)
+		test.That(tb, err, test.ShouldBeNil)
+		test.That(tb, linAcc, test.ShouldNotBeZeroValue)
+	})
+	accel, err := sensor.LinearAcceleration(context.Background(), nil)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, accel.X, test.ShouldEqual, expectedAccelX)
+	test.That(t, accel.Y, test.ShouldEqual, expectedAccelY)
+	test.That(t, accel.Z, test.ShouldEqual, expectedAccelZ)
+}
+
+//nolint:dupl
+func TestAngularVelocity(t *testing.T) {
+	angVelMockData := make([]byte, 16)
+	// x-vel
+	angVelMockData[8] = 64
+	angVelMockData[9] = 0
+	expectedAngVelX := 125.0
+	// y-accel
+	angVelMockData[10] = 32
+	angVelMockData[11] = 0
+	expectedAngVelY := 62.5
+	// z-accel
+	angVelMockData[12] = 16
+	angVelMockData[13] = 0
+	expectedAngVelZ := 31.25
+
+	logger := golog.NewTestLogger(t)
+	cfg, deps := setupDependencies(angVelMockData)
+	sensor, err := NewMpu6050(context.Background(), deps, cfg, logger)
+	test.That(t, err, test.ShouldBeNil)
+	defer utils.TryClose(context.Background(), sensor)
+	testutils.WaitForAssertion(t, func(tb testing.TB) {
+		angVel, err := sensor.AngularVelocity(context.Background(), nil)
+		test.That(tb, err, test.ShouldBeNil)
+		test.That(tb, angVel, test.ShouldNotBeZeroValue)
```

Comment:
Why use `test.ShouldNotBeZeroValue` here and not check that it is or is close to the mock values you set above?

Reply:
this is a wait function to make sure that we don't test the value until the coroutine that updates the struct has run at least once. It'll block the test function until the angular velocity is a non-zero value, then check the accuracy of the actual angular velocity value below

Reply:
Consider putting that response in a comment in the code. If Kim didn't see that, someone else is going to miss it, too.